src/ui/components/popup.vue

```diff
@@ -97,8 +97,13 @@ export default Vue.extend({
       window.close();
     },
     toggleIsolation(): void {
-      this.app.preferences.isolation.active = !this.app.preferences.isolation
-        .active;
+      this.app.storage.isolation.active = !this.app.storage.isolation.active;
+      browser.runtime.sendMessage({
+        method: 'saveIsolation',
+        payload: {
+          isolation: this.app.storage.isolation,
+        },
+      });
```

Comment:
This is my take on addressing #433.  Here are my thoughts:
- popup still needs to maintain its local state, thus app.storage.isolation.active is toggled
- the Vue watcher is on the app object
- a value change anywhere in app (including both preferences and storage) triggers the watcher
- Vue docs (https://vuejs.org/v2/api/#vm-watch) warns about object value changes, making old/new identical
- So...  since root.ts cannot distinguish app and oldApp, I removed it from the flow.
- Call sendMessage more as an event handler, keeping it generic, and let isolation.ts handle the changes.
Open to thoughts of course.

Note: Keep preferences on the watcher model, since a simple event handler isn't practical for preferences.  If Vue allows it (I'm not an expert there), watching app.preferences might provide better control for future changes.

Reply:
I'd say this is a solid approach, even though I'd prefer a more generic solution, but I don't see how this could be sanely done with things changing in the storage.

> If Vue allows it (I'm not an expert there), watching app.preferences might provide better control for future changes.

AFAIK it's not possible to watch just a nested property. This would maybe need some refactoring to move preferences into their data property in the root instead.